homeassistant/components/obihai/__init__.py

```diff
@@ -16,3 +44,24 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
     return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+
+class ObihaiEntity:
+    """Obihai Entity base Class."""
+
+    def __init__(self, pyobihai: PyObihai, serial: str, service_name: str) -> None:
+        """Initialize monitor sensor."""
+        self._pyobihai = pyobihai
+        self._service_name = service_name
+        self._unique_id = f"{serial}-{self._service_name}"
+        self._name = f"{OBIHAI} {self._service_name}"
```

Comment:
I know you had indicated this should be auto-constructed, and we can still do that.  I didn't love the fact that it auto-constructs with the unique ID.  The names end up containing the serial number.  I'm totally willing to re-work this.

Reply:
You are expected to set `_attr_has_entity_name=True` and `_attr_device_info=(...)`

Once that is done the entity name will be combined with the device name.